src/main/java/com/google/devtools/build/lib/remote/http/HttpDownloadHandler.java

```diff
@@ -137,6 +142,7 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
     DownloadCommand cmd = (DownloadCommand) msg;
     out = cmd.out();
     path = constructPath(cmd.uri(), cmd.digest().getHash(), cmd.casDownload());
+    skipBytes = (int)cmd.offset();
```

Comment:
What happens if the long offset is larger than Integer.MAX_VALUE?

Would it make sense to store skipBytes as a long field and instead around line 105 set skipNow=Integer.MAX_VALUE if skipBytes > Integer.MAX_VALUE?

Reply:
Good catch! IIRC I chose `long` for `offset` because `contentLength` is also `long`, but then found that `content.readableBytes` is `int`, so had to convert. But, you're right that this is not the right way to do that conversion.
I've changed `skipBytes` to be `long` and moved the conversion into the branch where `skipBytes < readableBytes`. Since `readableBytes` is `int` it's safe to convert in that case.